transports/janus_websockets.c

```diff
@@ -790,17 +794,46 @@ static int janus_websockets_callback(struct libwebsocket_context *this,
 			}
 			if(!ws_client->destroy && !g_atomic_int_get(&stopping)) {
 				janus_mutex_lock(&ws_client->mutex);
-				/* Shoot all the pending messages first */
+				/* Check if we have a pending/partial write to complete first */
+				if(ws_client->partial_message && ws_client->partial_offset > 0
+						&& !ws_client->destroy && !g_atomic_int_get(&stopping)) {
+					int pending = strlen(ws_client->partial_message) - ws_client->partial_offset;
+					JANUS_LOG(LOG_VERB, "[WSS-%p] Completing pending WebSocket write (still need to write last %d/%zu bytes)...\n",
+						wsi, pending, strlen(ws_client->partial_message));
+					int sent = libwebsocket_write(wsi, (unsigned char *)(ws_client->partial_message + ws_client->partial_offset), pending, LWS_WRITE_TEXT);
```

Comment:
Looking at the lws_write code, from what I understood it seems we do not need to have the additional padding if we're continuing a partial write: https://github.com/warmcat/libwebsockets/blob/6d64539fcbab0143903a549430445ff039a3aa52/lib/output.c#L271
There's a flag called ```inside_frame``` that is reset if all the message was sent:
https://github.com/warmcat/libwebsockets/blob/6d64539fcbab0143903a549430445ff039a3aa52/lib/output.c#L491
If it's not, it goes to a different branch of the code, which I guess makes sure that only the bytes are sent. I'll have to do a better check later to see if that's indeed the case.